src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,590 @@
+
+        logging.debug("Start: Retrieving classAds using Condor Python XQuery")
+        try:
+            itobj = schedd.xquery("WMAgent_AgentName == %s" % classad.quote(self.agent),
+                                  ['ClusterId', 'ProcId', 'JobStatus', 'MATCH_EXP_JOBGLIDEIN_CMSSite'])
+        except:
+            logging.debug("Query to condor schedd failed in SimpleCondorPlugin")
+            return runningList, changeList, completeList
+        else:
+            logging.debug("Finish: Retrieving classAds using Condor Python XQuery")
+            jobInfo = {}
+            for jobAd in itobj:
+                gridId = "%s.%s" % (jobAd['ClusterId'], jobAd['ProcId'])
+                jobStatus = str(jobAd.get('JobStatus', 100))
+                if jobStatus not in SimpleCondorPlugin.stateMap():
+                    jobStatus = '100'
+                location = jobAd.get('MATCH_EXP_JOBGLIDEIN_CMSSite', None)
+                jobInfo[gridId] = (jobStatus, location)
+
+            logging.debug("SimpleCondorPlugin retrieved %s classAds from condor schedd", len(jobInfo))
+
+        # now go over the jobs and see what we have
+        for job in jobs:
+
+            # if the schedd doesn't know a job, consider it complete
+            # doing any further checks is not cost effective
+            if job['gridid'] not in jobInfo:
+                (newStatus, location) = ('4', None)
+            else:
+                (newStatus,location) = jobInfo[job['gridid']]
+
+            # check for status changes
+            if newStatus != job['status']:
+
+                # update location info for Idle->Running transition
+                if newStatus == '2' and job['status'] == '1':
+                    if location:
+                        job['location'] = location
+                        logging.debug("JobAdInfo: Job location for jobid=%i gridid=%s changed to %s", job['jobid'], job['gridid'], location)
+
+                job['status'] = newStatus
+                job['status_time'] = int(time.time())
+                logging.debug("JobAdInfo: Job status for jobid=%i gridid=%s changed to %s", job['jobid'], job['gridid'], job['status'])
+                changeList.append(job)
+
+            job['globalState'] = SimpleCondorPlugin.stateMap()[newStatus]
+
+            # stop tracking finished jobs
+            if job['globalState'] in [ 'Complete', 'Error' ]:
+                completeList.append(job)
+            else:
+                runningList.append(job)
+
+        logging.debug("SimpleCondorPlugin tracking : %i/%i/%i (Executing/Changing/Complete)",
+                      len(runningList), len(changeList), len(completeList))
+
+        return runningList, changeList, completeList
+
+    def complete(self, jobs):
+        """
+        Do any completion work required
+
+        In this case, look for a returned logfile
+        """
+
+        for job in jobs:
+
+            if job.get('cache_dir') is None or job.get('retry_count') is None:
```

Comment:
I can do that but these lines are quite useless as they are. If they ever trigger something seriously has gone wrong.